tests/testthat/test-renderer3-figure-loss-small-data.R

```diff
@@ -0,0 +1,186 @@
+      "Number of data to segment",
+      breaks=c(range(some.props$models), 10, 100))+
+    ylab("Proportion of models selected by linear penalty")+
+    geom_point(aes(
+      models, prop.selected),
+      shape=21,
+      size=4,
+      clickSelects="pid.chr",
+      alpha=0.7,
+      data=data.frame(some.props)),
+  data=ggplot()+
+    ggtitle("Selected data set and model")+
+    theme_bw()+
+    ##theme_animint(update_axes=c("x", "y"))+
+    xlab("Position on chromosome (mega bases)")+
+    ylab("logratio (approximate DNA copy number")+
+    geom_segment(aes(
+      beforeStart, mean,
+      xend=afterEnd, yend=mean),
+      showSelected=c("pid.chr", "changes"),
+      color="green",
+      data=data.frame(some.segs))+
+    geom_vline(aes(
+      xintercept=megabases),
+      showSelected=c("pid.chr", "changes"),
+      color="green",
+      chunk_vars="pid.chr",
+      linetype="dashed",
+      data=data.frame(some.breaks))+
+    geom_point(aes(
+      megabases, logratio),
+      shape=21,
+      fill=NA,
+      showSelected="pid.chr",
+      data=data.frame(some.nb)),
+  loss=ggplot()+
+    ggtitle("Loss values for selected data set")+
+    ylab("loss")+
+    xlab("changes")+
+    theme_bw()+
+    theme_animint(update_axes=c("x", "y"))+
+    scale_size_manual(values=c(yes=3, no=4))+
+    geom_text(aes(
+      models, max.loss, label=paste0(
+        n.selected, "/", models,
+        " models selected by linear penalty")),
+      showSelected="pid.chr",
+      hjust=1,
+      data=data.frame(some.props))+
+    geom_point(aes(
+      changes, loss, color=selected, size=selected),
+      shape=21,
+      fill=NA,
+      showSelected="pid.chr",
+      data=data.frame(some.loss))+
+    geom_tallrect(aes(
+      xmin=min.changes, xmax=max.changes),
+      data=data.frame(some.loss),
+      alpha=0.5,
+      showSelected="pid.chr",
+      clickSelects="changes"),
+  lines=ggplot()+
+    ggtitle("Cost functions for selected data set")+
+    theme_bw()+
+    theme_animint(update_axes=c("x", "y"))+
+    geom_point(aes(
+      penalty, loss, color=selected),
+      fill=NA,
+      size=4,
+      showSelected="pid.chr",
+      data=data.frame(some.loss))+
+    geom_point(aes(
+      min.lambda, cost.at.min.lambda),
+      showSelected="pid.chr",
+      fill=NA,
+      data=data.frame(some.selection))+
+    ylab("cost(penalty) = loss + penalty*changes")+
+    geom_abline(aes(
+      slope=changes, intercept=loss, color=selected),
+      size=1,
+      showSelected="pid.chr",
+      chunk_vars="pid.chr",
+      data=data.frame(some.loss))+
+    geom_abline(aes(
+      slope=changes, intercept=loss, color=selected),
+      size=2,
+      showSelected=c("pid.chr", "changes"),
+      chunk_vars="pid.chr",
+      data=data.frame(some.loss)) 
+)
+info <- animint2HTML(viz)
+
+test_that("5 <circle> rendered for all data sets", {
+  dataSets=getNodeSet(info$html, '//g[@class="geom1_point_selected"]//g//circle')
+  expect_equal(length(dataSets), 5)
```

Comment:
can these expected numbers (5 and 72) please be changed from literals to `nrow(some.table)` ? (would make intent clearer I think)

Reply:
Thanks for the suggestion to improve the readability ! I rewrote the checks such that they are of the form nrow(data set)